backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -0,0 +1,78 @@
+package com.redhat.cloud.notifications.recipients.rbac;
+
+import com.redhat.cloud.notifications.recipients.RecipientUsersProvider;
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.routers.models.Page;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RbacRecipientUsersProvider implements RecipientUsersProvider {
+
+    @Inject
+    @RestClient
+    RbacServiceToService rbacServiceToService;
+
+    @ConfigProperty(name = "recipient-provider.rbac.elements-per-page", defaultValue = "40")
+    Integer rbacElementsPerPage;
+
+    @Override
+    public Uni<List<User>> getUsers(String accountId, boolean adminsOnly) {
```

Comment:
When will this be called? Since we occasionally have response time issues while calling RBAC, I am wondering if there is a chance this new service could slow us down even more.

Can we cache the methods results in this class?

Reply:
This will be called before sending an email, so we can have things like:
- `SendEmail: to all registered users` -> Should not be called
- `SendEmail: to all admins` -> Needs to call getUsers(accountId, true) and then filters by the registered users
- `SendEmail: to all admins (override)` -> Calls getUsers(accountId, true) but does not filter by the registered users
- `SendEmail: to all users (override)` -> Calls getUsers(accountId, false) - No filter
- `SendEmail: to all users of group Foo` -> getGroupUsers(....) - Filters
- etc

Not sure about caching, this won't take part of any http call

Reply:
Maybe we could cache the result after all.